### ISSUE TITLE
Flanneld: error registering network: cannot find network with management IP

### ISSUE THREAD

**Issue by ducttapecoder-vt** [coreos/flannel#1109]
This is a bit of a cross-post from my [stackoverflow ](https://stackoverflow.com/questions/55032935/flanneld-error-registering-network-cannot-finding-network-with-management-ip) and [Microsoft Forum](https://social.msdn.microsoft.com/Forums/en-US/aee7af97-679c-4a0d-8702-e4c42294d7aa/flanneld-error-registering-network-cannot-find-network-with-management-ip?forum=windowscontainers) questions since I'm not getting any traction there.

I'm setting up a mixed-mode kubernetes cluster (CentOS7 master, WS2019 worker) using Flannel in overlay (VXLAN) mode. I've gotten through the Microsoft [Kubernetes for Windows](https://docs.microsoft.com/en-us/virtualization/windowscontainers/kubernetes/joining-windows-workers?tabs=ManagementIP) Instructions but when I kick off start.ps1 I'm stuck in the "Waiting for the Network to be created" loop referenced [here](https://docs.microsoft.com/en-us/virtualization/windowscontainers/kubernetes/common-problems). Launching flanneld directly per their instructions, I get the following error:

    E0306 16:43:21.218797 2576 main.go:289] Error registering network: Cannot find network with Management IP [IPAddrofWorkerNIC].

The IP referenced is the main IP of the worker on the "Ethernet" NIC as called per the --iface argument to flanneld.

The master and worker are both Hyper-v VMs off of a Win10 1809 box with MAC spoofing enabled. I confirmed that 6433/tcp, 10250/tcp, 4096/udp, and 4789/udp are opened in firewalld of the master. I also tried again after disabling firewalld, with no change, so I don't think the issue is on the master side.

I tried digging through the flanneld Go code at the referenced line 289 for clues but I'm not familiar with Go and had to concede defeat. Google only comes back with the github repo and my stackoverflow question for responses to the error.

Any ideas why I would be getting this error?

## Your Environment
* Flannel version: flanneld.exe = v0.11.0
* Backend used (e.g. vxlan or udp): vxlan
* Etcd version: Whatever kubeadm deploys for kubenetes 1.13? Not sure how to determine this
* Kubernetes version (if used): 1.13
* Operating System and version: Win 10 1809 (host), WS 2019 1809 (Hyper-V VM, worker), CentOS 7 (Hyper-V VM, master)

**Comment by ducttapecoder-vt** [comment]
I didn't read the requirements closely enough. Apparently Flannel VXLAN on windows isn't supported until Kubernetes 1.14. Will try again after March 25.

```
Overlay networking: use Flannel in vxlan mode to configure a virtual overlay network

-     requires either Windows Server 2019 with KB4482887 installed or Windows Server vNext Insider Preview Build 18317+
-     requires Kubernetes v1.14 (or above) with WinOverlay feature gate enabled
-     requires Flannel v0.11.0 (or above)
```

**Comment by celestialorb** [comment]
I'm running into this as well with kubernetes 1.14.0-beta.2 so I'm not sure it's related to the version.

Seems to be related to this: https://github.com/coreos/flannel/blob/e4deb05e97ceb68d43df022fb76cd505a38a6973/backend/vxlan/vxlan_windows.go#L137

Since I'm not seeing `Checking HNS network for DR MAC...` my guess is that this line:
https://github.com/coreos/flannel/blob/e4deb05e97ceb68d43df022fb76cd505a38a6973/backend/vxlan/vxlan_windows.go#L125

is return an empty set of HNS networks.

**Comment by ducttapecoder-vt** [comment]
@celestialorb Interesting. Did you enable the WinOverlay feature gate?
I also had a similar issue with switching from overlay to host-gw because the script didn't clean up after itself. Ripping all the scripts and files out and running `Get-HNSNetwork | Remove-HNSNetwork` helped.

**Comment by celestialorb** [comment]
If I recall correctly that was the issue. I had a typo in my script that caused the `KubeletFeatureGates` parameter to be left out of the start-up script.

**Comment by Larswa** [comment]
I tried with the kubernetes 1.14.0-beta.2 executables on windows (1.13.4 on master), wanting to get it working with VXLAN mode for flannel. 

I also updated win2019 with the kb mentioned and added this parameter to start.ps1  `-KubeletFeatureGates "WinOverlay=true"` but got the same errors as @ducttapecoder-vt  Not sure if I got the syntax right for the featuregate?

**Comment by celestialorb** [comment]
@Larswa Syntax looks correct to me. Are you positive the update took? You can check to see if it exists with `Get-Hotfix` in Powershell. You probably have to restart the node after applying it?

**Comment by Larswa** [comment]
@celestialorb I'm fairly positive.  Windows update presented me with this updatehttps://support.microsoft.com/en-us/help/4489899 that bumped me to windows 1809 build 17763.379     And the earlier KB4482887 is build 17763.348 So I ought to have it.  

I amd working on a vagrant setup that can wipe and restore the clyster so I'll try again ... I had some manual not scripted things I tried, just to verify that I haven't fat fingered anything along the way .... waiting for my vagrant up now ...

**Comment by celestialorb** [comment]
I was able to get the error to disappear once installing this update (KB4482887): http://download.windowsupdate.com/c/msdownload/update/software/updt/2019/02/windows10.0-kb4482887-x64_826158e9ebfcabe08b425bf2cb160cd5bc1401da.msu

**Comment by Larswa** [comment]
I'll try with just THAT update on my next round ... thanks @celestialorb. Stay tuned.

**Comment by celestialorb** [comment]
Also make sure to restart, it wouldn't work for me until I restarted the machine after applying the update.

**Comment by pcdavies** [comment]
I applied the patch, but I continued to get the error. However, having had this problem in previous releases, I placed a "Start-Sleep 10" just after the "# Start kubelet" comment line and before the "$startKubeletArgs = ...." line in the start.ps1 script. Everything worked great after that. Once things finally worked, I was able to remove that additional sleep command, and things still work when running start.ps1. So I'm not positive if that was the ultimate fix, but worth a try.
